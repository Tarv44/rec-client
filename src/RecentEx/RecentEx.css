.recent li {
    margin: .5em 0;
    font-size: 20px;
}

.recent {
    align-items: flex-start;
    border-bottom: solid #F0E118 3px;
}