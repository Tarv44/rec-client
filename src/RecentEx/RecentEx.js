import { NavLink } from 'react-router-dom'

export default function RecentEx(props) {
    const recentEx = props.exchanges.sort((a,b) => {
        return a.date_created - b.date_created
    }).slice(0,3)

    const recentLis = recentEx.map(ex => {
        return <li key={ex.id}><NavLink to={`/exchange/${ex.id}`}>{ex.title}</NavLink></li>
    })

    return (
        <section>
            <NavLink to={'/add-exchange'}><button>Create New Exchange</button></NavLink>
            <h2>Recent Exchanges</h2>
            <ul>
                {recentLis}
            </ul>
        </section>
    )
}