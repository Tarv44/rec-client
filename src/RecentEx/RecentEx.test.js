import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import RecentEx from './RecentEx'
import store from '../dummy-store'
import { BrowserRouter } from 'react-router-dom'


describe('----- RecentEx -----', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render((
            <BrowserRouter>
                <RecentEx exchanges={store.exchanges}/>
            </BrowserRouter> 
        ), div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(
                <BrowserRouter>
                    <RecentEx exchanges={store.exchanges}/>
                </BrowserRouter> 
            )
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})