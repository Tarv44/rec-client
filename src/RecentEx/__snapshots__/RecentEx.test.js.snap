// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- RecentEx ----- renders the UI as expected 1`] = `
<section
  className="dashboard-section recent"
>
  <h2>
    Recently Updated
  </h2>
  <ul>
    <li>
      <a
        aria-current={null}
        href="/exchange/1"
        onClick={[Function]}
      >
        The Roots
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/exchange/2"
        onClick={[Function]}
      >
        Kiwi/Aussie Rock
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/exchange/3"
        onClick={[Function]}
      >
        Bark bark woof
      </a>
    </li>
  </ul>
</section>
`;
